llvm/include/llvm/SYCLLowerIR/GlobalOffset.h

```diff
@@ -15,11 +15,51 @@
 #ifndef LLVM_SYCL_GLOBALOFFSET_H
 #define LLVM_SYCL_GLOBALOFFSET_H
 
-#include "llvm/Pass.h"
+#include "llvm/IR/Module.h"
+#include "llvm/IR/PassManager.h"
+#include "llvm/SYCLLowerIR/TargetHelpers.h"
 
 namespace llvm {
 
-ModulePass *createGlobalOffsetPass();
+class ModulePass;
+class PassRegistry;
+
+class GlobalOffsetPass : public PassInfoMixin<GlobalOffsetPass> {
+private:
+  using KernelPayload = TargetHelpers::KernelPayload;
+  using ArchType = TargetHelpers::ArchType;
+
+public:
+  explicit GlobalOffsetPass() {}
+
+  PreservedAnalyses run(Module &M, ModuleAnalysisManager &);
+  static StringRef getPassName() { return "Add implicit SYCL global offset"; }
+
+private:
+  void processKernelEntryPoint(Module &M, Function *Func);
+  void addImplicitParameterToCallers(Module &M, Value *Callee,
+                                     Function *CalleeWithImplicitParam);
```

Comment:
Yeap, you could get the module from either of the function, seems like the function's interface hasn't changed since initial commit: https://github.com/intel/llvm/commit/c7bb28890c92618b4c91ca010c6a2751ff64ed1a#diff-6d821824806a585275ba835bcfb9180b42e55e1f54d8e09debdb16a6d10870d3R146 
Removing it now.